codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/CodegenVisitor.kt

```diff
@@ -229,12 +239,28 @@ class CodegenVisitor(context: PluginContext, private val codegenDecorator: RustC
         }
     }
 
+    /**
+     * Union Shape Visitor
+     *
+     * Generate an `enum` for union shapes.
+     *
+     * Note: this does not generate serializers
+     */
     override fun unionShape(shape: UnionShape) {
         rustCrate.useShapeWriter(shape) {
             UnionGenerator(model, symbolProvider, it, shape).render()
         }
     }
 
+    /**
+     * Generate service-specific code for the model:
+     * - Serializers
+     * - Deserializers
+     * - Fluent client
+     * - Trait implementations
+     * - Protocol tests
+     * - Operation structures
```

Comment:
Operation Rust structs are generated here or in `fn operationShape`?

Reply:
AFAIK in both places. I still need to dig into the `serviceShape()` generation. I have copied this comment from the client codegen and I will improve it once I understand this properly.